src/L1/rollup/L1MessageQueueWithGasPriceOracle.sol

```diff
@@ -59,14 +74,23 @@ contract L1MessageQueueWithGasPriceOracle is L1MessageQueue, IL1MessageQueueWith
      * Public View Functions *
      *************************/
 
+    /// @inheritdoc IL1MessageQueueWithGasPriceOracle
+    function l2BaseFee() public view returns (uint256) {
+        L2BaseFeeParameters memory parameters = l2BaseFeeParameters;
+        // this is unlikely to happen, use unchecked here
+        unchecked {
+            return (block.basefee * parameters.scalar) / PRECISION + parameters.overhead;
```

Comment:
we may set a back log task to decommission part of the gas-oracle as a reminder, the updating l2 base fee in l1 contract part.